File: mobile/src/main/java/xyz/mechenbier/circuittester/PowerStateAudio.java
package xyz.mechenbier.circuittester;

import android.content.Context;
import android.media.MediaPlayer;

/**
 * Created by tgrannen on 5/15/2016.
 */
public class PowerStateAudio {

  private MediaPlayer mp;
  public Context parentContext;
  private boolean Muted;
  private boolean SoundOnPowered = true;
  public boolean isCharging;

  public void AlterAudioState() {
    if (isCharging) {
      if (SoundOnPowered) {
        StartMP();
      } else {
        StopMP();
      }

    } else {
      if (SoundOnPowered) {
        StopMP();
      } else {
        StartMP();
      }
    }
  }

  public void SetSoundOnPowered(boolean isSoundOnPowered) {
    SoundOnPowered = isSoundOnPowered;
    AlterAudioState();
  }

  public void SetMuted(boolean isMuted) {
    Muted = isMuted;
    if (Muted) {
      StopMP();
    } else {
      AlterAudioState();
    }
  }

  public void InitMP() {
    if (parentContext == null) {
      return;
    }
    mp = MediaPlayer.create(parentContext, R.raw.test_alert);
    mp.setLooping(true);
  }

  public void InitMP(Context context) {
    parentContext = context;
    InitMP();
  }

  private void StartMP() {
    if (mp == null) {
      InitMP();
    }
    if (mp != null && !mp.isPlaying() && !Muted) {
      InitMP();
      mp.start();
    }
  }

  private void StopMP() {
    if (mp != null && mp.isPlaying()) {
      mp.stop();
      mp.reset();
    }
  }
}

File: mobile/src/main/java/xyz/mechenbier/circuittester/PowerStateAudio.kt
package xyz.mechenbier.circuittester

import android.content.Context
import android.media.MediaPlayer

/**
 * Created by tgrannen on 5/15/2016.
 * Updated by pmechenbier on 10/9/2018.
 */
class PowerStateAudio {

    private var mMediaPlayer: MediaPlayer? = null
    private var mMuted: Boolean = false
    private var mSoundOnPowered = true
    private var mParentContext: Context? = null
    var isCharging: Boolean? = null

    fun alterAudioState() {
        if (isCharging != null){
            if (isCharging!!) {
                if (mSoundOnPowered) {
                    startMP()
                } else {
                    stopMP()
                }

            } else {
                if (mSoundOnPowered) {
                    stopMP()
                } else {
                    startMP()
                }
            }
        }
    }

    fun setSoundOnPowered(isSoundOnPowered: Boolean) {
        mSoundOnPowered = isSoundOnPowered
        alterAudioState()
    }

    fun setMuted(isMuted: Boolean) {
        mMuted = isMuted
        if (mMuted) {
            stopMP()
        } else {
            alterAudioState()
        }
    }

    fun initMP(context: Context) {
        mParentContext = context
        initMP()
    }

    private fun initMP() {
        if (mParentContext == null) {
            return
        }
        mMediaPlayer = MediaPlayer.create(mParentContext, R.raw.test_alert)
        mMediaPlayer!!.isLooping = true
    }

    private fun startMP() {
        if (mMediaPlayer == null) {
            initMP()
        }
        if (mMediaPlayer != null && !mMediaPlayer!!.isPlaying && !mMuted) {
            initMP()
            mMediaPlayer!!.start()
        }
    }

    private fun stopMP() {
        if (mMediaPlayer != null && mMediaPlayer!!.isPlaying) {
            mMediaPlayer!!.stop()
            mMediaPlayer!!.reset()
        }
    }
}